### ISSUE TITLE
Download function enhancements

### ISSUE THREAD

**Issue by mitchellmanware** [Spatiotemporal-Exposures-and-Toxicology/NRTAPmodel#189]

**Comment by mitchellmanware** [comment]
Function enhancements will:

1. Remove redundant `cat(paste0()); stop()` when returning error
2. Add `unzip = ` argument for functions that download zip files
  1. Allows user to retain only zip files to unzip all at once
3.

**Comment by mitchellmanware** [comment]
## Ongoing unit test development (12/14)
- [ ] download_aqs_data()
- [ ] download_koppen_geiger_data()
- [ ] download_merra2_data()
- [ ] download_modis_data()
- [ ] download_sedac_groads_data()
- [ ] download_sedac_population_data()

**Comment by eva0marques** [comment]
Hi @mitchellmanware,

I was taking a look at your download functions unit tests. It's really cool to have tests for them!
I was just wondering if we would not do better to store the test_download_functions function in the tests/testthat folder. Actually I was even wondering if it is a common practice to write functions for tests.

**Comment by sigmafelix** [comment]
@mitchellmanware @eva0marques 
FYI, I tested writing files in temporary folders using `tempdir()`. It made no writing permission errors. Perhaps applying this would help to improve coverage of download functions (i.e., download_run() parts).

**Comment by eva0marques** [comment]
Yes my point was about the practice to create test support functions in the R folder.

**Comment by sigmafelix** [comment]
@mitchellmanware 
I believe so. Please refer to my test example through the link below:

https://github.com/Spatiotemporal-Exposures-and-Toxicology/Scalable_GIS/blob/2dd2c715c4264fd7b1ab3d3cbf08319fb7e87e70/tests/testthat/tests.R#L874-L946

`tempdir()` function seems to be dependent on the current session.

**Comment by mitchellmanware** [comment]
@eva0marques 
I do not know it creating functions for the tests is common practice. I decided to function-ize the repeated steps in the tests similar to the way we did for the download functions themselves.

**Comment by mitchellmanware** [comment]
## December 18, 2023
The `input/Rinput/download_functions/download.R` file can be moved to the `R/` folder after implementing at least the URL check unit tests. See above comments for potential expansion to executing some data download using temporary directories with `tempdir()`.

**Comment by mitchellmanware** [comment]
Function enhancements were pulled into the main branch via [pull request ](https://github.com/Spatiotemporal-Exposures-and-Toxicology/NRTAPmodel/pull/232) from branch kpm-pkgdown-01 (December 22, 2023).
